script.js

```diff
@@ -1,29 +1,285 @@
+        keyNote: 'e4',
+        keyCode: 'KeyH',
+        label: 'H',
+        pressed: false,
+        frequency: 329.6276
+    },
+    {
+        keyColour: 'white',
+        keyNote: 'f4',
+        keyCode: 'KeyJ',
+        label: 'J',
+        pressed: false,
+        frequency: 349.2282,
+    },
+    {
+        keyColour: 'black',
+        parentKey: 'f4',
+        keyNote: 'f4sharp',
+        keyCode: 'KeyI',
+        label: 'I',
+        pressed: false,
+        frequency: 369.9944
+    },
+    {
+        keyColour: 'white',
+        keyNote: 'g4',
+        keyCode: 'KeyK',
+        label: 'K',
+        pressed: false,
+        frequency: 391.9954
+    },
+    {
+        keyColour: 'black',
+        parentKey: 'g4',
+        keyNote: 'g4sharp',
+        keyCode: 'KeyO',
+        label: 'O',
+        pressed: false,
+        frequency: 415.3047
+    },
+    {
+        keyColour: 'white',
+        keyNote: 'a5',
+        keyCode: 'KeyL',
+        label: 'L',
+        pressed: false,
+        frequency: 440.0000
+    },
+    {
+        keyColour: 'black',
+        parentKey: 'a5',
+        keyNote: 'a5sharp',
+        keyCode: 'KeyP',
+        label: 'P',
+        pressed: false,
+        frequency: 466.1638
+    },
+    {
+        keyColour: 'white',
+        keyNote: 'b5',
+        keyCode: 'Semicolon',
+        label: ';',
+        pressed: false,
+        frequency: 493.8833
+    },
+    {
+        keyColour: 'white',
+        keyNote: 'c5',
+        keyCode: 'Quote',
+        label: "'",
+        pressed: false,
+        frequency: 523.2511
+    },
+    {
+        keyColour: 'black',
+        parentKey: 'c5',
+        keyNote: 'c5sharp',
+        keyCode: 'BracketLeft',
+        label: '[',
+        pressed: false,
+        frequency: 554.3653
+    },
+    {
+        keyColour: 'white',
+        keyNote: 'd5',
+        keyCode: 'Backslash',
+        label: '\\',
+        pressed: false,
+        frequency: 587.3295
+    },
+    {
+        keyColour: 'black',
+        parentKey: 'd5',
+        keyNote: 'd5sharp',
+        keyCode: 'BracketRight',
+        label: ']',
+        pressed: false,
+        frequency: 622.2540
+    },
+    {
+        keyColour: 'white',
+        keyNote: 'e5',
+        keyCode: 'Enter',
+        label: '⏎',
+        pressed: false,
+        frequency: 659.2551,
+    },
+]
+
+keyBoardArray.forEach((key) => {
+    if(key.keyColour === 'white') {
+        pianoKeys.innerHTML +=
+        '<div id="' + key.keyNote + '" class="piano-key white-key" data-id="' + key.keyNote + '">\n' +
+        '<p>' + key.label + '</p>\n' +
+        '</div>'
+    }
+})
+
+keyBoardArray.forEach((key) => {
+    if(key.keyColour === 'black') {
+        let parentWhiteKey = document.querySelector('#' + key.parentKey )
+        parentWhiteKey.innerHTML +=
+            '<div id="' + key.keyNote + '" class="piano-key black-key" data-note="' + key.keyNote + '">\n' +
+            '   <p>' + key.label + '</p>\n' +
+            '</div>\n'
     }
```

Comment:
should be only 1 foreach loop because this loops twice over the same array, thus wasting half of the loop iterations (as discussed)